perf/roles/trex/tasks/main.yml

```diff
@@ -9,29 +9,47 @@
       - python3
       - rdma-core-devel
       - zlib-devel
+      - driverctl
     state: installed
   tags: install
 
+- name: Install kernel-devel
+  yum:
+    name: kernel-devel-{{ ansible_kernel }}
+    state: present
+  tags: install
+
+- name: Install Development Tools group
+  yum:
+    name: "@Development Tools"
+    state: present
+  tags: install
+
 - name: stop existing trex server
   command: pkill -x _t-rex-64
   ignore_errors: true
   tags: setup
 
-- name: clone trex
-  git:
-    repo: '{{trex_repo}}'
-    dest: '{{trex_dir}}'
-    version: '{{trex_tag}}'
-    force: true
+- name: Create TRex install directory
+  file:
+    path: "{{ trex_install_dir }}"
+    state: directory
+    mode: "0755"
+  tags: install
+
+- name: Download TRex release
+  get_url:
+    url: "{{ trex_release_url }}"
+    dest: "{{ trex_install_dir }}/{{ trex_release_filename }}"
+    validate_certs: no
+    force: yes
   tags: install
 
-- name: configure and build trex
-  command:
-    cmd: './b {{item}}'
-    chdir: '{{trex_dir}}/linux_dpdk'
-  loop:
-    - configure
-    - build
+- name: Extract TRex release
+  unarchive:
+    src: "{{ trex_install_dir }}/{{ trex_release_filename }}"
+    dest: "{{ trex_install_dir }}"
+    copy: no
```

Comment:
Instead of download + unarchive, what about using a remote source?

https://docs.ansible.com/ansible/latest/collections/ansible/builtin/unarchive_module.html#parameter-src

Reply:
Actually, that's better. Updated this one also.